src/spec/validitygenerator.py

```diff
@@ -413,7 +413,15 @@ def createValidationLineForParameterIntroChunk(self, blockname, param, params, t
                 asciidoc += 'a pointer to '
 
             if paramtype.text == 'void':
-                # If there's only one pointer, it's optional, and it doesn't point at anything in particular - we don't need any language.
+                self.logMsg('error',
+                            ' Parameter ' + blockname + '::' + paramname.text + ' is a ' + paramtype.text + paramtype.tail.strip() + '.'
+                            ' void* without noautovalidity and without len attribute is a source of errors.'
+                            ' It has been prohibited.'
+                            ' Likely you forgot len or forgot noautovalidity attribute in vk.xml.'
+                            ' In the rare case this is true pointer to nothing or opaque memory,'
+                            ' you are now required to add noautovalidity too.')
```

Comment:
Do we really need the error? For other cases where it's ambiguous the autovalidity script simply doesn't attempt to do anything with it. If we ever add a function or struct, people will look at the built spec, see what's missing, and add it - works for other parameter types, think it's probably fine for this? I'd probably rather just not handle this type of parameter and silently do nothing. If we add an error for this then there might be an expectation for other such error checks which simply do not exist right now.

Reply:
That's too psychologically and sociologically deep for me...
I guess I could make it into inline comment and in the error message say something cryptic like "Parsing error - invalid registry" to preserve the image of an unreliable script.

> people will look at the built spec, see what's missing

I think you have too much trust in people though. Half of the formating bugs wouldn't be there in the first place if people did so. E.g. care to explain how the broken blocks below `oldSwapchain` happened if people built the spec and evaluated what is missing? No offence meant to anybody...

Reply:
> I guess I could make it into inline comment and in the error message say something cryptic like "Parsing error - invalid registry" to preserve the image of an unreliable script.

I don't think that's a good idea. If you're going to throw an error, say something useful. I just don't see the value in this particular error - you're basically just forcing users to awkwardly acknowledge that there's no VU for void* parameters, by throwing an error and telling them to put "noautovalidity" on it.

If you drop the "noautovalidity" requirement and drop any sort of handling for void* parameters, I don't think anything really changes. The main benefit of this change IMO is not dumping out misleading/incorrect VU statements - which is what it currently does.

> I think you have too much trust in people though. Half of the formating bugs wouldn't be there in the first place if people did so. E.g. care to explain how the broken blocks below oldSwapchain happened if people built the spec and evaluated what is missing? No offense meant to anybody...

Oh no they definitely screw it up - it's just that in this case the worst possible outcome of not throwing an error is "oh I missed a VU", which is trivially fixable later. We definitely need more sanity checks for other things though, but that's a whole other issue..

Reply:
> If you drop the "noautovalidity" requirement and drop any sort of handling for void* parameters, I don't think anything really changes.

Yeah, well, it is the raison d'etre of this PR...

Alternativelly implicit validity for `void*` could be redefined to, well, have no implicit validity. That would result in no VU statements generated and no `noautovalidity` needed (even in #559).
I simply liked the `noautovalidity` version better, because it implies there must be explicit VU. And there must be explicit VU. And unlike any other type, `void*` only on implicit validity is almost sure to be an error (except for pass-through pointers like `pUserData`).

> error is "oh I missed a VU", which is trivially fixable later.

Fixable and fixed are two different things.
PR #559 is evidence that those kinds of human errors do happen, and also that their authors never fix them before release(, or at least in timely manner anyway). (And it fixes those errors.)
This PR strives to make sure that the same error never happens in the future.

Reply:
> Alternativelly implicit validity for void* could be redefined to, well, have no meaningful implicit validity. That would result in no VU statements generated and no noautovalidity needed (even in #559 cases). That what you proposing?

Yes, precisely that.

> Fixable and fixed are two different things.

Sure, but we don't add void* parameters to the spec very often anyway (it's largely discouraged). Adding additional VU constraints is more about clarifying the situation for the validation layers rather than changing any behaviour - it's always been invalid, but the VU will make it obvious that's the case. If we were at 100% coverage for existing VUs I'd be more worried, but as it stands there's probably a few hundred VU statements missing from the spec in various edge cases - this won't really help with that wider problem.

Reply:
> Yes, precisely that.

Very well then, I will prepare such variant.

> If we were at 100% coverage for existing VUs I'd be more worried, but as it stands there's probably a few hundred VU statements missing from the spec in various edge cases - this won't really help with that wider problem.

Ew, hundreds? Oh my!

Reply:
Yea I could probably pick a random function, stare at it for an hour and find a chunk of things that need to be there. I just... don't want to do that :D

Reply:
I wish I had your discipline. You are describing precisely what tends to happen to me. :D 

Hmm, if I get rid of `noautovalidity`, should `optional` still have (non-NULL) meaning? In that case the VU section would be spammed with redundant non-NULL statements. Which would force authors to use `noautovalidity` there anyway.
Damit, I am doing it right now. 😱 Ok I am gonna need to stare at it for an hour 😅

Reply:
> I wish I had your discipline. You are describing precisely what tends to happen to me. :D

Haha yea I can't say I hadn't noticed ;)

> Hmm, if I get rid of noautovalidity, should optional still have (non-NULL) meaning? In that case the VU section would be spammed with redundant non-NULL statements. Which would force authors to use noautovalidity there anyway.
Damit, I am doing it right now. 😱 Ok I am gonna need to stare at it for an hour 😅

Ewww... Ugh damnit yea now I'm going to have to go stare at them too :P!

Reply:
@TobiasHector I am gonna go with true implicit validity for `pUserData`, and `noautovalidity` elsewhere. If future author does not add `noautovalidity`, there would be no harm except VU saying "non-NULL" redundant to his own explicit VU statement.

Reply:
@TobiasHector Yeah, I am gonna go with true implicit for `pUserData`, indirect optional for `void**` in mapMemory and `noautovalidity` elsewhere. There's not much harm if future author screws it up -- it would only result in benign non-NULL implicit VU redundant to his own explicit VU.